### ISSUE TITLE
Steam runtime does not build

### ISSUE THREAD

**Issue by dkudriavtsev** [NixOS/nixpkgs#59851]
## Issue description

The Steam runtime has been updated, so the old packages are unable to be fetched and return 404 errors.

### Steps to reproduce

Attempt to install Steam while on the unstable channel.

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.0.7, NixOS, 19.09pre175916.1fc591f9a5b (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.09pre175916.1fc591f9a5b"`
 - channels(dmitry): `"nixos-19.09pre173080.1222e289b50"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by shirona** [comment]
Seconding this. In addition, it seems Steam won't run at all due to an outdated `steamrt`, so an update is urgent.
```
mkdir: cannot create directory ‘/nix/store/nvkirkk91ak1n33039jc3fabvz1scqm3-steam-fhs/steamrt/pinned_libs_32’: Read-only file system
mkdir: cannot create directory ‘/nix/store/nvkirkk91ak1n33039jc3fabvz1scqm3-steam-fhs/steamrt/pinned_libs_64’: Read-only file system
internal error: /steamrt/setup.sh is missing, this runtime is invalid or corrupted
```

**Comment by Slabity** [comment]
Can confirm...

Starting to get a bit annoyed at this breaking every single time the runtime updates. Is there no way to automate this?

**Comment by sio-k** [comment]
I have the same problem, and get the exact same output as shirona above.

system details:
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.34, NixOS, 19.09pre175916.1fc591f9a5b (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.09pre175916.1fc591f9a5b"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by shirona** [comment]
@slabity
> Can confirm...
> 
> Starting to get a bit annoyed at this breaking every single time the runtime updates. Is there no way to automate this?
> 
> EDIT: Looks like #59838 fixes it

Yes, that seems like it. Sadly, nixos-unstable is two days old yet. It usually doesn't take all that long to rebuild, so hopefully we will be able to update tomorrow.
https://howoldis.herokuapp.com/

**Comment by jakubdaniel** [comment]
There already is an updated package in https://github.com/NixOS/nixpkgs/pull/59822

**Comment by Slabity** [comment]
@jakubdaniel - That's a backport of a fix for earlier this week when it broke before. This is the second time it broke.

**Comment by cpages** [comment]
@shirona & @sio-k the setup.sh missing issue is separate and is already fixed both in master and 19.03 (see https://github.com/NixOS/nixpkgs/pull/59326), but it'll take a bit until it reaches the channels. Steam seems to run just fine even without the updated runtime from #59838

**Comment by cpages** [comment]
#59838 is also in. I'm leaving this open until the channels catch up
